### ISSUE TITLE
Make an option for a responsive table

### ISSUE THREAD

**Issue by MartinDawson** [palantir/blueprint#4443]
#### Environment

- __Package version(s)__:     
    "@blueprintjs/core": "^3.36.0",
    "@blueprintjs/table": "^3.8.18",
- __Browser and OS versions__: 
    Chrome latest
    Windows 1

#### Feature request

It's a really fundamental feature. The table should be responsive, i.e. it should take up 100% of the width of the screen and then can shrink down to certain sizes for tablet/mobile.

#### Examples

Similar to this one: https://github.com/palantir/blueprint/issues/969

But that was closed and is years old.

**Comment by adidahiya** [comment]
Can you please be more specific about what "responsive" means for you? Do you expect the columns to expand to fill as much space as possible, like a flex layout?

I can mark this as a feature request but it is low priority for us at this time and is likely a large undertaking.

**Comment by MartinDawson** [comment]
@adidahiya Yeah exactly, so the problem right now is that the table does not wisely use up 100% of the width available to it if it has the space for it.

Here's an example, you can see there are some ghost cells at the end column.
![image](https://user-images.githubusercontent.com/15030491/103926311-a84edf80-5110-11eb-9a34-3214d90ab107.png)

It would be great if we could have an option to stretch the columns wider if the space allows for it.

If the space doesn't allow for it (i.e mobile devices) then the table should just shrink to a minWidth per column.

If this is possible in the current version, it would be great to see some example code for how I can implement it (not in the library) but as it stands I don't think I can.

Thanks for the great package and work!

**Comment by adidahiya** [comment]
We added ghost cells for this purpose, to make the behavior more like a spreadsheet rather than an HTML table. It's not a simple feature to make the columns stretch and work with all the layout calculations, as I noted above.

**Comment by lanzhenw** [comment]
@adidahiya  Can we allow the ghost cells to share the same style as the custom headers?

<img width="649" alt="Screen Shot 2021-02-19 at 11 53 59 AM" src="https://user-images.githubusercontent.com/17770824/108544357-f3c9e100-72ab-11eb-878b-92e94b7a7049.png">

**Comment by jgentes** [comment]
FWIW, I used this simple workaround to at least maximize the table column widths based on the page/container width:

```
const docWidth = (document.getElementById('tableId')?.clientWidth || 1000) - 30 // row header is 30px

const colWidths = {
      col1: 0.3,
      col2: 0.1,
      col3: 0.1,
      col4: 0.1,
      col5: 0.1,
      col6: 0.2,
      col7: 0.1
    }

const columnDefs = [{
   title: 'col1',
   width: docWidth * colWidths.col1
}]
```

Then when defining the table: 

`<Table columnWidths={columnDefs.map(c => c.width)} />`

This workaround doesn't help when resizing the window unless you force a re-render, which would then require de-bouncing the re-render method to avoid too many renders while resizing.

Would definitely be nice to see variable width supported out of the box, even if the table simply allowed percentage values rather than just numbers for column width.
